src/hooks/useHandleAppTileEvents.ts

```diff
@@ -0,0 +1,108 @@
+import { Linking } from 'react-native';
+import type {
+  WebViewMessageEvent,
+  WebViewNavigation,
+} from 'react-native-webview';
+import { useAppConfig } from './useAppConfig';
+import { useNavigation } from '@react-navigation/native';
+import { HomeStackParamList } from '../navigators/types';
+import { StackNavigationProp } from '@react-navigation/stack';
+
+type NavigationParams = {
+  [x: string]: unknown | undefined;
+  referenceDate?: string | number | Date;
+};
+
+type DeepLinkRouteName = 'tiles/Today/Survey';
+
+export enum AppTileMessageType {
+  deepLink = 'deepLink',
+  openUrl = 'openUrl',
+  openConsent = 'openConsent',
+}
+
+export type DeepLinkDataType = {
+  routeName: DeepLinkRouteName;
+  params: NavigationParams;
+};
+
+export type OpenUrlDataType = {
+  url: string;
+  title?: string;
+  appendUtm?: boolean;
+};
+
+export type DeepLinkAppletMessage = {
+  type: AppTileMessageType.deepLink;
+  data: DeepLinkDataType;
+};
+
+export type OpenUrlAppletMessage = {
+  type: AppTileMessageType.openUrl;
+  data: OpenUrlDataType;
+};
+
+export type AppTileMessage = DeepLinkAppletMessage | OpenUrlAppletMessage;
+
+export const useHandleAppTileEvents = () => {
+  const { data } = useAppConfig();
+  const todayTileSettings = data?.homeTab?.todayTileSettings;
+  const navigation = useNavigation<StackNavigationProp<HomeStackParamList>>();
+
+  const openSurveyAppTile = (responseId: string | undefined) => {
+    if (!todayTileSettings?.surveysTile || !responseId) {
+      return;
+    }
+    navigation.push('Home/AuthedAppTile', {
+      appTile: todayTileSettings.surveysTile,
+      searchParams: { responseId },
+    });
+  };
+
+  const handleDeepLinkMessage = (appletMessage: DeepLinkAppletMessage) => {
+    const { routeName, params = {} } = appletMessage.data;
+    if (routeName === 'tiles/Today/Survey') {
+      openSurveyAppTile(params.questionnaire as string | undefined);
+    } else {
+      console.warn('Unsupported route name. Ignoring...', { routeName });
+    }
+  };
+
+  const enforceWebURLProtocol = (url: string) => {
+    const allowedProtocols = ['https://', 'http://'];
```

Comment:
Any reason you know of why we need to support `http://`.  If you're not sure, it'd be great to start this initial migration with `https://` being the only option.

Reply:
Oh, I see - this is used by `handleOpenUrlMessage`, which I'm betting is for program actions that like a youtube URL, article to review, or journal/info website?  Got it if so - if we allow program configuration with either scheme, probably best to keep both allowed here as well.  It's forcibly non-authed URLs here, which is key.

Reply:
That's correct. This is for URLs entered by the admin in the program builder pointing to articles to read, etc. We do not currently enforce a particular protocol.